packages/EasyCodingStandard/src/Console/Output/JsonOutputFormatter.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\EasyCodingStandard\Console\Output;
+
+use Nette\Utils\Json;
+use Symplify\EasyCodingStandard\Configuration\Configuration;
+use Symplify\EasyCodingStandard\Console\Style\EasyCodingStandardStyle;
+use Symplify\EasyCodingStandard\Contract\Console\Output\OutputFormatterInterface;
+use Symplify\EasyCodingStandard\Error\Error;
+use Symplify\EasyCodingStandard\Error\ErrorAndDiffCollector;
+use Symplify\EasyCodingStandard\Error\FileDiff;
+use Symplify\PackageBuilder\Console\ShellCode;
+
+final class JsonOutputFormatter implements OutputFormatterInterface
+{
+    /**
+     * @var EasyCodingStandardStyle
+     */
+    private $easyCodingStandardStyle;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var ErrorAndDiffCollector
+     */
+    private $errorAndDiffCollector;
+
+    public function __construct(
+        EasyCodingStandardStyle $easyCodingStandardStyle,
+        Configuration $configuration,
+        ErrorAndDiffCollector $errorAndDiffCollector
+    ) {
+        $this->easyCodingStandardStyle = $easyCodingStandardStyle;
+        $this->configuration = $configuration;
+        $this->errorAndDiffCollector = $errorAndDiffCollector;
+    }
+
+    public function report(int $processedFilesCount): int
+    {
+        $errorsArray = [
+            'totals' => [
+                'errors' => $this->errorAndDiffCollector->getErrorCount(),
+                'diffs' => $this->errorAndDiffCollector->getFileDiffsCount(),
+            ],
+            'files' => [],
+        ];
+
+        /** @var Error[] $errors */
```

Comment:
Is this typehint needed? It should be in `getErrors()` return type already

Reply:
For some reason, PHPStorm is not happy. It marks `errors` as `any`.

Reply:
I see. No troubles, keep it